### DISCUSSION TITLE
Next.js / SCSS Modules - not working as expected (with repro)

### DISCUSSION THREAD

**Discussion by chris-erickson** [storybookjs/storybook#25081] [Help]
### Summary

Using the latest storybook 7.6.2 with the NextJS add-on. But I get tons of errors like this:

```
./src/components/Sharing/ShareBlock.stories.js 6:15-25
export 'ShareBlock' (imported as 'ShareBlock') was not found in './ShareBlock' (possible exports: default)
```

Which seems to me like the modules aren't being converted from `ShareBlock.module.scss` properly. They're referenced like:

```javascript
import { social } from './ShareBlock.module.scss';
```

These of course result in empty styles like:
```html
<div class>Hello</div>
```

I do nothing but the default setup of NextJS for styles that I'm aware.

I'm looking at [this section of the blog post](https://storybook.js.org/blog/integrate-nextjs-and-storybook-automatically/#styling) or [this section of the longer documentation](https://github.com/storybookjs/storybook/tree/next/code/frameworks/nextjs#csssassscss-modules).

I don't know if it's relevant, but if I use the default Webpack builder I get these errors in the storybook UI:
```
TypeError: [BABEL] /Development/website/packages/site/.storybook/preview.js: _lruCache is not a constructor (While processing: "base$0")
```
Using the swc builder has none of that (and I use the SWC builder for the NextJS site already) so I'll stick with it.

### Additional information

```javascript
import path from 'path';
import { join, dirname } from 'path';

/**
 * This function is used to resolve the absolute path of a package.
 * It is needed in projects that use Yarn PnP or are set up within a monorepo.
 */
function getAbsolutePath(value) {
  return dirname(require.resolve(join(value, 'package.json')));
}

/** @type { import('@storybook/nextjs').StorybookConfig } */
const config = {
  stories: ['../src/**/*.mdx', '../src/**/*.stories.@(js|jsx|mjs|ts|tsx)'],
  addons: [
    getAbsolutePath('@storybook/addon-links'),
    getAbsolutePath('@storybook/addon-essentials'),
    getAbsolutePath('@storybook/addon-interactions'),
  ],
  framework: {
    name: getAbsolutePath('@storybook/nextjs'),
    options: {
      builder: { useSWC: true },
    },
  },
  staticDirs: ['../public'],
  docs: {
    autodocs: 'tag',
  },
};
export default config;
```

### Chromatic

https://main--6569ff9a7ba13f719dd0dc34.chromatic.com

### Create a reproduction

https://github.com/chris-erickson/next-js-scss-modules-repro

**Comment by valentinpalkovic** [comment]
I took a quick look at https://nextjs.org/docs/app/building-your-application/styling/css-modules, I can see that style is imported per default import and then you can access the class name you want.

```tsx
import styles from './styles.module.css'
 
export default function DashboardLayout({
  children,
}: {
  children: React.ReactNode
}) {
  return <section className={styles.dashboard}>{children}</section>
}
```

Can you point me to some official Next.js documentation that shows, that named exports can be imported as well?

**Reply by chris-erickson** [reply]
That's about the extent of NextJS docs on the issue, but we use this _everywhere_, like dozens or hundreds of components:

```
import {
  header,
  logo,
  keyNav,
  extraItem,
  logoMobile,
  logoWide,
  navModal,
  modalOverlay,
  modalButton,
  divider,
} from './header.module.scss';
```

If the solution is _you can't, and need to rewrite all of them_ I guess I'm ok with it but definitely weird that we can do it in Next but not Storybook.

**Comment by valentinpalkovic** [comment]
I see! I guess it's some webpack configuration we are missing here. I try to find out what the cause of this different behavior is. Can you tell me, whether it is working as expected in Babel mode? Just set the `useSWC` flag to false.

**Reply by chris-erickson** [reply]
There's no difference in Babel mode (same result).

**Reply by chris-erickson** [reply]
Do you have a sense of whether this is a quick fix or something that will need time and sequencing to get looked at? Wondering if it's a thing I can pitch in on somehow too, but my SB internals knowledge is ... zero. 😄

**Reply by valentinpalkovic** [reply]
I haven't had any time yet to take a look, but feel free to dive deep into the code and figure it out. If you have any questions, I am happy to assist.

**Reply by chris-erickson** [reply]
Would love if you could direct me to where this might be configured or where the good place to start looking is. Not sure I have time, might almost be able to convert these all to default imports rather than named faster..
